### PR TITLE
Refactor/get journey injectable middleware

### PR DESCRIPTION
**Pull request by Si3rr4wow**
~This one still needs work (and some significant rebasing) but the idea is~ the journeys are now injectable

### CODE REVIEW

File: packages/forms-web-app/src/dynamic-forms/journey-factory.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 const { JOURNEY_TYPES } = require('@pins/common/src/dynamic-forms/journey-types');
-const { buildJourneyParams: buildHASLPAJourneyParams } = require('./has-questionnaire/journey');
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/journey-factory.js:2]: most of the code in here ends up redundant, it was just easier for my brain to port it so I could keep track of what I was changing

File: packages/forms-web-app/src/dynamic-forms/journeys.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		return Object.keys(this.journeyParams);
+	}
+
+	/**
+	 * @param {JourneyResponse} journeyResponse
+	 * @returns {Journey}
+	 */
+	getJourney(journeyResponse) {
+		const { journeyId } = journeyResponse;
+		console.log('🚀 ~ Journeys ~ getJourney ~ journeyId:', journeyId);
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/journeys.js:33]: 🚮

File: packages/forms-web-app/src/dynamic-forms/middleware/dynamic-req-files-to-req-body-files.js [hunk 3]
```diff
@@ -19,9 +17,8 @@ module.exports = () => (req, res, next) => {
 	}
 
 	const { section, question } = req.params;
-	const journeyResponse = res.locals.journeyResponse;
+	const { journey } = res.locals.journeyResponse;
```
Comment (Si3rr4wow) [inline packages/forms-web-app/src/dynamic-forms/middleware/dynamic-req-files-to-req-body-files.js:20]:
```suggestion
	const { journey } = res.locals;
```

File: packages/forms-web-app/src/dynamic-forms/journey-factory.js (hunk 1 above)
Comment (Si3rr4wow) [reply packages/forms-web-app/src/dynamic-forms/journey-factory.js:2]: removed all the redundant bits now

**Review by bryn-pins — COMMENTED** [review]
think it's shaping up nicely, looks like an issue during merge though?

File: packages/forms-web-app/src/dynamic-forms/controller.js [hunk 4]
```diff
@@ -279,34 +273,12 @@ exports.submit = async (req, res) => {
 	);
 };
 
-// Submit LPA Statement
-/**
- * @type {import('express').Handler}
- */
-exports.submitLpaStatement = async (req, res) => {
```
[the remark this answers is not in the document]
Comment (Si3rr4wow) [reply packages/forms-web-app/src/dynamic-forms/controller.js:286]: yep good spot!

File: packages/forms-web-app/src/dynamic-forms/controller.js [hunk 5]
```diff
@@ -438,8 +408,22 @@ exports.lpaSubmitted = async (req, res) => {
  * @type {import('express').Handler}
  */
 exports.appellantSubmitted = async (req, res) => {
```
[the remark this answers is not in the document]
Comment (Si3rr4wow) [reply packages/forms-web-app/src/dynamic-forms/controller.js:430]: 👍

**Review by bryn-pins — DISMISSED** [review]

File: packages/forms-web-app/src/journeys/index.js
Comment (bryn-pins) [inline packages/forms-web-app/src/journeys/index.js:1]: this has changed in rebase, we missing some registered journeys?

Comment (Si3rr4wow) [reply packages/forms-web-app/src/journeys/index.js:1]: Dang it

Comment (Si3rr4wow) [reply packages/forms-web-app/src/journeys/index.js:1]: readded. praying it's not updating that new journey that's making this unmergable

**Pull request merged**